src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/ingredient/UniqueIngredientList.java

```diff
@@ -71,6 +72,36 @@ public void setIngredients(List<Ingredient> ingredients) {
         internalList.setAll(ingredients);
     }
 
+    /**
+     * Replaces the similar ingredient that is in the address book with a new ingredient whose quantity is reduced
+     * by the quantity times the multiplier in {@code target} if it exists.
+     *
+     * @param target The target ingredient.
+     * @param multiplier The multiplier.
+     */
+    public void minusIngredientQuantity(Ingredient target, Quantity multiplier) {
+        requireAllNonNull(target, multiplier);
+
+        int index = -1;
+        Ingredient similarToTarget = null;
+        for (int i = 0; i < internalList.size(); i++) {
+            Ingredient currentIngredient = internalList.get(i);
+            if (currentIngredient.isSameIngredient(target)) {
+                index = i;
+                similarToTarget = currentIngredient;
+                break;
+            }
+        }
+
+        if (index == -1) {
+            return;
+        }
+
+        internalList.set(index, new Ingredient(similarToTarget.getName(),
+                        similarToTarget.getQuantity().minusQuantityBy(target.getQuantity(), multiplier),
+                        similarToTarget.getUnit()));
+    }
```

Comment:
Is this more readable?
```suggestion
    public void minusIngredientQuantity(Ingredient target, Quantity multiplier) {
        requireAllNonNull(target, multiplier);
        
      Ingredient currentIngredient = internalList.stream()
      .filter(ingredient -> ingredient.isSameIngredient(target))
      .findFirst()
      .orElse(null);

      if (currentIngredient != null) {
          Ingredient ingredientWithNewQuantity = new Ingredient(
                  currentIngredient.getName(),
                  currentIngredient.getQuantity().minusQuantityBy(target.getQuantity(), multiplier),
                  currentIngredient.getUnit());

          int index = internalList.indexOf(currentIngredient);
          internalList.set(index, ingredientWithNewQuantity);
    }
```